### DISCUSSION TITLE
k3s_control_node_address & k3s_etcd_datastore

### DISCUSSION THREAD

**Discussion by mbwmbw1337** [PyratLabs/ansible-role-k3s#98] [General]
If using a k3s_etcd_datastore (true), is k3s_control_node_address used? My understanding is when etcd is used, there is no need for a load-balancer in front of the control node. 

Would you be able to provide clarification on that?

**Comment by xanmanning** [comment]
Hi.

`k3s_control_node_address` serves as a [fixed registration address for agent nodes](https://rancher.com/docs/k3s/latest/en/architecture/#fixed-registration-address-for-agent-nodes) when using embedded etcd. This can be either a load balancer or a round robin DNS record.

This is the same for if you use an external datastore, the concept is the same, a single address that spans a cluster of server nodes so that worker nodes can register with the control plane.

It is not required in this role, the role will default to set this to the first server node in the control plane.

Regardless of topology, a highly available control plane would benefit from this being set to the address of a load balancer or round robin DNS record so that if one server node dies, you can communicate with another available server node.
